File: app/src/main/java/com/worldventures/dreamtrips/social/ui/tripsimages/model/converter/BaseVideoConverter.java
package com.worldventures.dreamtrips.social.ui.tripsimages.model.converter;

import com.innahema.collections.query.queriables.Queryable;
import com.worldventures.core.converter.Converter;
import com.worldventures.dreamtrips.social.ui.feed.model.video.Quality;
import com.worldventures.dreamtrips.social.ui.feed.model.video.Video;

import io.techery.mappery.MapperyContext;

public abstract class BaseVideoConverter<T extends com.worldventures.dreamtrips.api.multimedia.model.Video> implements Converter<T, Video> {
   @Override
   public Class<Video> targetClass() {
      return Video.class;
   }

   @Override
   public Video convert(MapperyContext mapperyContext, T videoAttachment) {
      Video video = new Video();
      if (videoAttachment.ratio() != null) {
         video.setAspectRatio(videoAttachment.ratio());
      }
      if (videoAttachment.qualities() != null) {
         video.setQualities(Queryable.from(videoAttachment.qualities())
               .map(it -> new Quality(it.quality(), it.uri())).toList());
      }
      video.setThumbnail(videoAttachment.thumbnail());
      video.setUid(videoAttachment.uid());
      video.setUploadId(videoAttachment.uploadId());
      video.setCreatedAt(videoAttachment.createdAt());
      video.setDuration(videoAttachment.duration());
      return video;
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/social/ui/tripsimages/model/converter/BaseVideoConverter.kt
package com.worldventures.dreamtrips.social.ui.tripsimages.model.converter

import com.worldventures.core.converter.Converter
import com.worldventures.dreamtrips.social.ui.feed.model.video.Quality
import com.worldventures.dreamtrips.social.ui.feed.model.video.Video
import io.techery.mappery.MapperyContext

typealias ApiVideo = com.worldventures.dreamtrips.api.multimedia.model.Video

abstract class BaseVideoConverter<T : ApiVideo> : Converter<T, Video> {
   override fun targetClass(): Class<Video> = Video::class.java

   override fun convert(mapperyContext: MapperyContext, videoAttachment: T): Video {
      val video = Video()
      var videoAttachmentRatio = videoAttachment.ratio()
      if (videoAttachmentRatio != null) {
         video.aspectRatio = videoAttachmentRatio
      }
      val videoQualities = videoAttachment.qualities()
      if (videoQualities != null) {
         video.qualities = videoQualities.map { it -> Quality(it.quality(), it.uri()) }.toList()
      }
      video.thumbnail = videoAttachment.thumbnail()
      video.uid = videoAttachment.uid()
      video.uploadId = videoAttachment.uploadId()
      video.createdAt = videoAttachment.createdAt()
      video.duration = videoAttachment.duration()
      return video
   }
}
